db/dcrpg/pgblockchain.go

```diff
@@ -4827,6 +4827,19 @@ func (pgb *ChainDB) GetAPITransaction(txid *chainhash.Hash) *apitypes.Tx {
 		return nil
 	}
 
+	msgTx, err := txhelpers.MsgTxFromHex(txraw.Hex)
+	if err != nil {
+		log.Errorf("Cannot create MsgTx for tx %v: %v", txid.String(), err)
+		return nil
+	}
+
+	treasuryActive := true
+	if txraw.BlockHeight > 0 {
+		treasuryActive = txhelpers.IsTreasuryActive(pgb.chainParams.Net, txraw.BlockHeight)
+	}
+
+	txType := txhelpers.DetermineTxType(msgTx, treasuryActive)
```

Comment:
tx type != tree.

I think you're after `txhelpers.TxTree`: https://github.com/decred/dcrdata/blob/5d2df4b7f43f7b937515f9ca79dc79f911e2a3cd/txhelpers/txhelpers.go#L1202-L1209

which itself uses `DetermineTxType`: https://github.com/decred/dcrdata/blob/5d2df4b7f43f7b937515f9ca79dc79f911e2a3cd/txhelpers/txhelpers.go#L1197-L1200

Reply:
Oh, we just want this tree to be either `stake` or `regular`.

Reply:
`type` as in string?

Reply:
Yeah.  There's a `txhehlpers.TxTypeToString` helpers.  We might want to pass that through `strings.ToLower` for JSON, or even make a `api/types.TxType type` with a stringer and the reverse.